test/e2e/app/app.go

```diff
@@ -17,14 +18,20 @@ import (
 	"github.com/cometbft/cometbft/abci/example/kvstore"
 	abci "github.com/cometbft/cometbft/abci/types"
 	"github.com/cometbft/cometbft/crypto"
+	cryptoenc "github.com/cometbft/cometbft/crypto/encoding"
 	"github.com/cometbft/cometbft/libs/log"
+	"github.com/cometbft/cometbft/libs/protoio"
+	cryptoproto "github.com/cometbft/cometbft/proto/tendermint/crypto"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
 	"github.com/cometbft/cometbft/version"
 )
 
 const (
 	appVersion                 = 1
 	voteExtensionKey    string = "extensionSum"
 	voteExtensionMaxVal int64  = 128
+	reservedKey         string = "reservedTxKey"
```

Comment:
Eventhough addresses do not contain `y`, it might be a good idea to add a separator to the end of the key.
```suggestion
	reservedKey         string = "reservedTxKey"
```

Reply:
Yeah, good idea

Reply:
Also, comming in the next PR